components/movementsensor/adxl345/adxl345.go

```diff
@@ -189,10 +266,113 @@ func NewAdxl345(
 			}
 		}
 	})
-
+	sensor.interruptsFound = make(map[string]interruptInfo)
+	sensor.readInterrupts(sensor.cancelContext)
+	err = sensor.configureInterruptRegisters(ctx)
+	if err != nil {
+		sensor.cancelFunc()
+		return nil, err
+	}
+	if sensor.echoInterrupt1 != nil {
+		sensor.startInterruptPolling(sensor.echoInterrupt1)
+	}
+	if sensor.echoInterrupt2 != nil {
+		sensor.startInterruptPolling(sensor.echoInterrupt2)
+	}
 	return sensor, nil
 }
 
+func (adxl *adxl345) startInterruptPolling(interrupt board.DigitalInterrupt) {
+	utils.PanicCapturingGo(func() {
+		interrupt.AddCallback(adxl.ticksChan)
+		defer interrupt.RemoveCallback(adxl.ticksChan)
+
+		for {
+			select {
+			case <-adxl.cancelContext.Done():
+				return
+			case tick := <-adxl.ticksChan:
+				if tick.High {
+					func() {
+						adxl.mu.Lock()
+						defer adxl.mu.Unlock()
+						adxl.readInterrupts(adxl.cancelContext)
+					}()
+				}
+			}
+		}
+	})
+}
+
+func getInterruptConfigurations(int1, int2 InterruptPinAttrConfig) map[byte]byte {
```

Comment:
this is unnecessary if each interrupt has it's own channel

Reply:
Could you explain this a little more? This function is to get the values that need to be written to the registers related to turning on and mapping interrupts to the correct pins. What does the channel help with?